### DISCUSSION TITLE
How to just download specification of a gem without caching or updating user directories

### DISCUSSION THREAD

**Discussion by nareshdevasani** [ruby/rubygems#6612] [Q&A]
Dear team,

I need help to find a mechanism to download a gem along with its specification.
Command used:
`gem install prm --no-user-install`
Current problem is that it is accessing the cache directory which is at the user's home dirrectory.

Another command is:
`gem specification prm`

But all of these are doing something in the cache directory which in some cases don't have permissions to access:
```
$ gem env
RubyGems Environment:
  - RUBYGEMS VERSION: 3.4.10
  - RUBY VERSION: 3.2.2 (2023-03-30 patchlevel 53) [x86_64-linux]
  - INSTALLATION DIRECTORY: /usr/local/rvm/gems/ruby-3.2.2
  - USER INSTALLATION DIRECTORY: $HOME/.local/share/gem/ruby/3.2.0
  - RUBY EXECUTABLE: /usr/local/rvm/rubies/ruby-3.2.2/bin/ruby
  - GIT EXECUTABLE: /usr/bin/git
  - EXECUTABLE DIRECTORY: /usr/local/rvm/gems/ruby-3.2.2/bin
  - SPEC CACHE DIRECTORY: $HOME/.local/share/gem/specs
```

what I need is to simply download the gem with specification or just the specification of a gem to a given directory.

Thanks for your help.

**Comment by deivid-rodriguez** [comment]
Hi. You can set `GEM_SPEC_CACHE` to a folder you can write to. Hope this helps!
